frontend/src/features/workspace/components/CreateFormModal/CreateFormModalContent/SaveSecretKeyScreen.tsx

```diff
@@ -64,250 +49,22 @@ const useSaveSecretKeyDefault = () => {
-                </Text>
-              </ListItem>
-            </UnorderedList>
-            <Stack direction={{ base: 'column', md: 'row' }}>
-              <Tooltip
-                mt={0}
-                label={t(
-                  `features.workspace.modals.create.secretKey.tooltip.${hasCopiedKey ? 'copied' : 'copyKey'}`,
-                )}
-              >
-                <Code
-                  // To allow for focus styling on code element.
-                  data-group
-                  tabIndex={0}
-                  flex={1}
-                  transition="background 0.2s ease"
-                  cursor="pointer"
-                  onClick={handleCopyKey}
-                  _groupFocus={{
-                    bg: 'neutral.400',
-                  }}
-                  _hover={{
-                    bg: 'neutral.300',
-                  }}
-                  wordBreak="break-word"
-                  display="inline-flex"
-                  alignItems="center"
-                  w="100%"
-                  h="auto"
-                  px="0.75rem"
-                  py="0.625rem"
-                  bg="neutral.200"
-                  color="secondary.500"
-                  borderRadius="4px"
-                >
-                  {secretKey}
-                </Code>
-              </Tooltip>
-              <ButtonGroup>
-                <Button onClick={handleDownloadKey}>
-                  {t('features.workspace.modals.create.secretKey.download')}
-                </Button>
-                <IconButton
-                  as="a"
-                  icon={<BiMailSend />}
-                  aria-label="Email the secret key to someone"
-                  href={mailToHref}
-                  variant="outline"
-                />
-              </ButtonGroup>
-            </Stack>
-          </Box>
-          {hasDownloaded && (
-            <Box mt="1rem">
-              <Checkbox
-                aria-label="Storage mode form acknowledgement"
-                {...register('storageAck', {
-                  required: true,
-                })}
-              >
-                {t('features.workspace.modals.create.secretKey.declaration')}
-              </Checkbox>
-            </Box>
-          )}
-          <Button
-            mt="2.25rem"
-            isDisabled={!isSubmitEnabled}
-            rightIcon={<BiRightArrowAlt fontSize="1.5rem" />}
-            type="submit"
-            isLoading={isLoading}
-            onClick={handleDownloadAndNavigate}
-            isFullWidth
-          >
-            <Text lineHeight="1.5rem">
-              {t('features.workspace.modals.create.secretKey.confirm')}
-            </Text>
-          </Button>
-        </Container>
-      </ModalBody>
-    </>
+    <SaveSecretKeyContent
+      contentTitle={t('features.workspace.modals.create.secretKey.title')}
+      registerStorageAck={
+        // HACK: Since CreateFormWizardInputProps must have storageAck, we can cast it to SaveSecretKeyFormInput.
```

Comment:
The CreateFormWizardInputProps contains > fields than SaveSecretKeyFormInput 

SaveSecretKeyFormInput only requires `storageAck`, but having issues resolving the type due to the UseFormRegister

Reply:
Open to any suggestions on a cleaner way to resolve this typescript issue apart from typecasting to `as unknown as ...`

Reply:
https://github.com/orgs/react-hook-form/discussions/8418